tls/s2n_ktls_io.c

```diff
@@ -210,3 +217,53 @@ S2N_RESULT s2n_ktls_sendmsg(struct s2n_connection *conn, uint8_t record_type, co
     *bytes_written = result;
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_ktls_recvmsg(struct s2n_connection *conn, uint8_t *record_type, uint8_t *buf,
+        size_t buf_len, s2n_blocked_status *blocked, size_t *bytes_read)
+{
+    RESULT_ENSURE_REF(bytes_read);
+    RESULT_ENSURE_REF(blocked);
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(buf);
+    /* Ensure that buf_len is > 0 since trying to receive 0 bytes does not
+     * make sense and a return value of `0` from recvmsg is treated as EOF.
+     */
+    RESULT_ENSURE_GT(buf_len, 0);
+
+    *blocked = S2N_BLOCKED_ON_READ;
```

Comment:
Oh, I should have commented on this in sendmsg too, but you should probably set `bytes_read` to 0 when you set blocked to S2N_BLOCKED_ON_READ. `bytes_read` should be ignored on error, but setting it to zero would help guard against caller mistakes.

Reply:
since we return record_type we prob also should `0` that out and therefore add a NULL check :)